design/one-pager-component-mutable-and-versioning.md

```diff
@@ -359,6 +359,34 @@ spec:
 
 Still, the `Traffic` trait should handle garbage collection things for the old workloads when no longer needed.
 
+#### Multiple revisions of one Component can exist together
+
+It's allowed to have multiple revisions of the same Component existing simultaneously.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: my-app-deployment
+  annotations:
+    version: v1.0.0
+    description: "Description of this deployment"
+spec:
+  components:
+    - revisionName: example-component-1
+      traits:
+        - apiversion:core.oam.dev/v1alpha2
+          kind: ManualScaler
+          spec:
+            replicaCount: 3
+    - revisionName: example-component-2
+      traits:
+        - apiversion:core.oam.dev/v1alpha2
+          kind: ManualScaler
+          spec:
+            replicaCount: 2
```

Comment:
We are considering it's an anti-pattern, as we could use two different components to do the same thing. Using different revisions of one component  seems don't bring any extra bonus.

Reply:
Okay, before the proposal is ready, let's suspend this PR. But I will separate e2e test covering current versioning mechanism into another PR as well as the fix of #170.

Reply:
Cool, we can hold this PR until we have a conclusion on that issue